src/plugins/core/settings.ts

```diff
@@ -26,6 +36,29 @@ export class SettingsPlugin extends CorePlugin {
     return this.locale;
   }
 
+  private changeCellsDateFormatWithLocale(oldLocale: Locale, newLocale: Locale) {
+    for (const sheetId of this.getters.getSheetIds()) {
+      for (const [cellId, cell] of Object.entries(this.getters.getCells(sheetId))) {
+        let formatToApply: Format | undefined;
+        if (cell.format === oldLocale.dateFormat) {
+          formatToApply = newLocale.dateFormat;
+        }
+        if (cell.format === getDateTimeFormat(oldLocale)) {
```

Comment:
shouldn't we also check `timeFormat` while we are at it?

Reply:
True, I changed that